src/praisonai-agents/praisonaiagents/memory/enhanced_memory.py

```
@@ -0,0 +1,647 @@
+        """Store user memory (legacy compatibility)."""
+        metadata = {"user_id": user_id}
+        if extra:
+            metadata.update(extra)
+        return self.store_short_term(text, metadata=metadata)
+    
+    def search_user_memory(self, user_id: str, query: str, limit: int = 5, **kwargs) -> List[Dict[str, Any]]:
+        """Search user memory (legacy compatibility)."""
+        results = self.search_short_term(query, limit=20)
+        return [r for r in results if r.get("metadata", {}).get("user_id") == user_id][:limit]
+    
+    def search(self, query: str, user_id: Optional[str] = None, agent_id: Optional[str] = None, 
```

Ruff (0.14.14)

[warning] Unused method argument: `agent_id`

(ARG002)